### PR TITLE
typography mdx

### PR DESCRIPTION
**Pull request by jonborchardt**

### CODE REVIEW

File: components/textStyles.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import styled, { css } from 'styled-components';
+
+const Jumbo = styled.span`
+    font-family: ${({ theme }) => theme.typography.textStyles.jumbo.fontFamily};
+    font-size: ${({ theme }) => theme.typography.textStyles.jumbo.fontSize};
+    line-height: ${({ theme }) => theme.typography.textStyles.jumbo.lineHeight};
```
Comment (codeviking) [inline components/textStyles.tsx:6]: I think now might be the time to remove the per font-size line height. Line-height should be a ratio, and I bet the one we set at the top-level (i.e. on the `body` or `html` element) is likely appropriate to use elsewhere.

WDYT?

File: components/textStyles.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    ${baseCodeStyles}
+
+    padding: ${({ theme }) => theme.typography.textStyles.code.padding};
+`;
+
+const InlineCode = styled.code<{ variant?: CodeVariant }>`
+    ${baseCodeStyles}
+
+    padding: 0 ${({ theme }) => theme.spacing.xxs};
+    margin: 0 ${({ theme }) => theme.spacing.xxs};
```
Comment (codeviking) [inline components/textStyles.tsx:77]: It'll need `display: inline-block` for these to apply. Do we set that somewhere?

File: demo/src/pages/Breakpoints/BreakpointValues.tsx [hunk 3]
```diff
@@ -3,14 +3,14 @@ import styled from 'styled-components';
...
-import { BodyMicro } from '@allenai/varnish/components';
+import { textStyles } from '@allenai/varnish/components';
 
 export class BreakpointValues extends React.PureComponent {
     render() {
         return (
             <>
                 <Namespace>
-                    <BodyMicro>theme.breakpoints.</BodyMicro>
+                    <textStyles.Micro>theme.breakpoints.</textStyles.Micro>
```
Comment (codeviking) [inline demo/src/pages/Breakpoints/BreakpointValues.tsx:13]: Should this be `TextStyles`, like we're doing elsewhere (or do we only do that when the root, i.e. `TextStyles` is a component unto itself)?

File: demo/src/pages/TextStyles/demo/code.mdx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+---
+order: 2
+title: Code
+---
+
+Code use Roboto Mono.
```
Comment (codeviking) [inline demo/src/pages/TextStyles/demo/code.mdx:6]:
```suggestion
The `<Code />` component uses a monospace font to help make the source code both legible and familiar.
```

File: demo/src/pages/TextStyles/demo/default.mdx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+---
+order: 0
+title: Default
+---
+
+Default usage.
+
+Text styles uses Lato Regular and Bold.
```
Comment (codeviking) [inline demo/src/pages/TextStyles/demo/default.mdx:8]:
```suggestion
The text style components can be used to render text using the default-font in a variety of sizes.  
```

Comment (codeviking) [reply demo/src/pages/TextStyles/demo/default.mdx:8]: I don't think our users care what the font-family is :).

File: demo/src/pages/TextStyles/demo/quote.mdx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+---
+order: 1
+title: Quote
+---
+
+Quotes and their authors use Volkhov Regular.
```
Comment (codeviking) [inline demo/src/pages/TextStyles/demo/quote.mdx:6]:
```suggestion
Use this component for rendering quoted text. This helps the quote stand out, which both helps the user distinguish what text is quoted and looks nice.
```

File: demo/src/pages/TextStyles/page.mdx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+const mdxs = require.context('./demo', false, /\.(mdx)$/);
+const examples = mdxs.keys().map(mdxs);
+
+## Text Styles
+
+Basic text styles.
+
+### When To Use
+
+Use when you wish to alter the style typographic elements.
```
Comment (codeviking) [inline demo/src/pages/TextStyles/page.mdx:18]:
```suggestion
Most of the time you won't need to use these components, as the text should be styled according to our defaults simply by using Varnish. That said you should use these components when you want to make the text larger or smaller, or if you're rendering a specific type of text, like source code or quoted material.
```

File: demo/src/pages/TextStyles/page.mdx [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+iconSrc: ./text-14px.svg
+---
+import defaultDemo from './demo/default.mdx'; // crazy, we must import for the require to work
+// these two lines grab the mdx files in the local demo dir and make examples for each
+const mdxs = require.context('./demo', false, /\.(mdx)$/);
+const examples = mdxs.keys().map(mdxs);
+
+## Text Styles
+
+Basic text styles.
```
Comment (codeviking) [inline demo/src/pages/TextStyles/page.mdx:14]: These components help you style text.

**Review by codeviking — COMMENTED** [review]
This looks pretty good, but I think we should probably drop per-variant `line-height` in favor of a single, top-level ratio attached to the `<body />`. Let me know what you think.

File: components/textStyles.tsx (hunk 1 above)
Comment (jonborchardt) [reply components/textStyles.tsx]: good idea, done

File: components/textStyles.tsx (hunk 2 above)
Comment (jonborchardt) [reply components/textStyles.tsx:69]: done

File: demo/src/pages/TextStyles/page.mdx (hunk 8 above)
Comment (jonborchardt) [reply demo/src/pages/TextStyles/page.mdx]: done

File: demo/src/pages/Breakpoints/BreakpointValues.tsx (hunk 3 above)
Comment (jonborchardt) [reply demo/src/pages/Breakpoints/BreakpointValues.tsx:13]: there is no TextStyles component
we only capitalize if there is

File: theme/typography.ts [hunk 9]
```diff
@@ -15,6 +15,9 @@ const font = {
         headline: `Lato,'Helvetica Neue',Helvetica,-apple-system,BlinkMacSystemFont,'Segoe UI','PingFang SC','Hiragino Sans GB','Microsoft YaHei',Arial,sans-serif,'Apple Color Emoji','Segoe UI Emoji','Segoe UI Symbol'`,
         code: `'Roboto Mono',SFMono-Regular,Consolas,'Liberation Mono',Menlo,Courier,monospace`,
     },
+    lineHeight: {
+        default: 1.375, // 22px/16px (chosen as average of previous non ratio values)
+    },
```
Comment (codeviking) [inline theme/typography.ts:20]: 👍

File: theme/typography.ts [hunk 10]
```diff
@@ -55,40 +58,28 @@ const h6 = {
     fontWeight: font.weight.regular,
 };
 
-// body
-const bodyJumbo = {
+// textStyles
+const jumbo = {
     fontFamily: font.family.sansSerif,
     fontSize: '24px',
     lineHeight: 36 / 24,
```
Comment (codeviking) [inline theme/typography.ts:65]: Should this be removed?

**Review by codeviking — APPROVED** [review]
LGTM; There's still some `lineHeight` references in `theme/typography.ts`, do we need those still or should they be removed (I don't think they're used anymore).

**Pull request merged**